modules/wallet/update.go

```diff
@@ -219,6 +223,10 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 // ReceiveUpdatedUnconfirmedTransactions updates the wallet's unconfirmed
 // transaction set.
 func (w *Wallet) ReceiveUpdatedUnconfirmedTransactions(txns []types.Transaction, _ modules.ConsensusChange) {
+	if err := w.tg.Add(); err != nil {
+		panic(err)
+	}
+	defer w.tg.Done()
```

Comment:
I'm pretty sure you can trigger this codepath anyway, it should just gracefully return. There's a race condition with how the consensus updates and tpool updates work.